cmd/cstor-pool-mgmt/controller/backup-controller/handler.go

```diff
@@ -0,0 +1,237 @@
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/tools/cache"
+)
+
+// syncHandler compares the actual state with the desired, and attempts to
+// converge the two. It then updates the Status block of the CStorReplicaUpdated resource
+// with the current status of the resource.
+func (c *BackupController) syncHandler(key string, operation common.QueueOperation) error {
+	glog.Infof("Sync handler called for key:%s with op:%s", key, operation)
+	bkp, err := c.getBackupCStorResource(key)
+	if err != nil {
+		return err
+	}
+	if bkp == nil {
+		return fmt.Errorf("cannot retrieve BackupCStor %q", key)
+	}
+	if IsDoneStatus(bkp) || IsFailedStatus(bkp) {
+		return nil
+	}
+
+	status, err := c.eventHandler(operation, bkp)
+	if status == "" {
+		return nil
+	}
+
+	if err != nil {
+		glog.Errorf(err.Error())
+		bkp.Status = apis.BKPCStorStatusFailed
+	} else {
+		bkp.Status = apis.BackupCStorStatus(status)
+	}
+
+	nbkp, err := c.clientset.OpenebsV1alpha1().BackupCStors(bkp.Namespace).Get(bkp.Name, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	nbkp.Status = bkp.Status
+
+	_, err = c.clientset.OpenebsV1alpha1().BackupCStors(nbkp.Namespace).Update(nbkp)
+	if err != nil {
+		return err
+	}
+
+	glog.Infof("Completed operation:%v for backup:%v, status:%v", operation, nbkp.Name, nbkp.Status)
+	return nil
+}
+
+// eventHandler will execute a function according to a given operation
+func (c *BackupController) eventHandler(operation common.QueueOperation, bkp *apis.BackupCStor) (string, error) {
+	switch operation {
+	case common.QOpAdd:
+		return c.addEventHandler(bkp)
+	case common.QOpDestroy:
+		/* TODO: Handle backup destroy event
+		 */
+		return "", nil
+	case common.QOpSync:
+		return c.syncEventHandler(bkp)
+	case common.QOpModify:
+		return "", nil
+	}
+	return string(apis.BKPCStorStatusInvalid), nil
+}
+
+// addEventHandler will change the state of backup to Init state.
+func (c *BackupController) addEventHandler(bkp *apis.BackupCStor) (string, error) {
+	if !IsPendingStatus(bkp) {
+		return string(apis.BKPCStorStatusInvalid), nil
+	}
+	return string(apis.BKPCStorStatusInit), nil
+}
+
+// syncEventHandler will perform the backup if a given backup is in init state
+func (c *BackupController) syncEventHandler(bkp *apis.BackupCStor) (string, error) {
+	// If the backup is in init state then only we will complete the backup
+	if IsInitStatus(bkp) {
+		bkp.Status = apis.BKPCStorStatusInProgress
+		_, err := c.clientset.OpenebsV1alpha1().BackupCStors(bkp.Namespace).Update(bkp)
+		if err != nil {
+			glog.Errorf("Failed to update backup:%s status : %v", bkp.Name, err.Error())
+			return "", err
+		}
+
+		err = volumereplica.CreateVolumeBackup(bkp)
+		if err != nil {
+			c.recorder.Event(bkp, corev1.EventTypeNormal, string(common.SuccessCreated), string(common.MessageResourceCreated))
```

Comment:
These message seem to be wrong?
Can we indent these lines so that logic is more readable

Reply:
ok